File: src/main/java/br/com/g1bet/controller/PartidaController.java
package br.com.g1bet.controller;

import br.com.g1bet.model.Partida;
import br.com.g1bet.repository.PartidaRepository;
import br.com.g1bet.service.PartidaService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/partida")
@CrossOrigin(origins = "*", allowedHeaders = "*")
public class PartidaController {

    @Autowired
    private PartidaService service;

    @Autowired
    private PartidaRepository repository;

    @GetMapping("/historico/{idTime}")
    public ResponseEntity<List<Partida>> exibirHistoricoPartida(@PathVariable Long idTime) {
        return ResponseEntity.ok(service.exibirHistoricoPartida(idTime));
    }

    @PostMapping
    public ResponseEntity<Partida> cadastrar(@RequestBody Partida partida) {
        return ResponseEntity.ok(service.cadastrar(partida));
    }

    @DeleteMapping("/{id}")
    public void delete(@PathVariable Long id) {
        service.deletar(id);
    }

    @PutMapping
    public ResponseEntity<Partida> put(@RequestBody Partida partidaModel) {
        return ResponseEntity.status(HttpStatus.OK)
                .body(repository.save(partidaModel));
    }

}

File: src/main/java/br/com/g1bet/controller/PartidaController.kt
package br.com.g1bet.controller

import br.com.g1bet.model.Partida
import br.com.g1bet.repository.PartidaRepository
import br.com.g1bet.service.PartidaService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/partida")
@CrossOrigin(origins = arrayOf("*"), allowedHeaders = arrayOf("*"))
class PartidaController {
    @Autowired
    private val service: PartidaService? = null

    @Autowired
    private val repository: PartidaRepository? = null
    @GetMapping("/historico/{idTime}")
    fun exibirHistoricoPartida(@PathVariable idTime: Long?): ResponseEntity<List<Partida>> {
        return ResponseEntity.ok(service!!.exibirHistoricoPartida(idTime))
    }

    @PostMapping
    fun cadastrar(@RequestBody partida: Partida?): ResponseEntity<Partida> {
        return ResponseEntity.ok(service!!.cadastrar(partida))
    }

    @DeleteMapping("/{id}")
    fun delete(@PathVariable id: Long?) {
        service?.deletar(id)
    }

    @PutMapping
    fun put(@RequestBody partidaModel: Partida?): ResponseEntity<Partida> {
        return ResponseEntity.status(HttpStatus.OK)
                .body(repository.save(partidaModel))
    }
}